mesonbuild/modules/java.py

```diff
@@ -17,60 +17,64 @@
 import typing as T
 from mesonbuild import mesonlib
 from mesonbuild.build import CustomTarget
-from mesonbuild.compilers import detect_compiler_for
+from mesonbuild.compilers import detect_compiler_for, Compiler
 from mesonbuild.interpreterbase.decorators import ContainerTypeInfo, FeatureDeprecated, FeatureNew, KwargInfo, typed_pos_args, typed_kwargs
 from mesonbuild.interpreter.interpreterobjects import FileHolder
 from mesonbuild.mesonlib import version_compare, MachineChoice
-from . import ExtensionModule, ModuleReturnValue, ModuleState
-from ..interpreter import Interpreter
+from . import NewExtensionModule, ModuleReturnValue, ModuleState
 
-class JavaModule(ExtensionModule):
+class JavaModule(NewExtensionModule):
     @FeatureNew('Java Module', '0.60.0')
-    def __init__(self, interpreter: Interpreter):
-        super().__init__(interpreter)
+    def __init__(self, *args: T.Any, **kwargs: T.Any):
```

Comment:
just `def __init__(self) -> None:` right? (need the None to tell mypy to type check the function

Reply:
I believe we still receive the interpreter here. Otherwise Python would error out, right?

Reply:
We shouldn't be using `T.Any` if we can avoid it, its a step in the wrong direction either way. Either leave it as-is, or remove the interpreter argument if we don't need it.